### ISSUE TITLE
ENOENT: no such file or directory - util.js

### ISSUE THREAD

**Issue by FHoxhaDev** [modesty/pdf2json#330]
I'm facing this issue below when using my application in vercel. It works fine on localhost but it doesnt on vercel.

Error:

```
Error: ENOENT: no such file or directory, open '/var/task/node_modules/pdf2json/base/shared/util.js'
    at Object.openSync (node:fs:596:3)
    at Object.readFileSync (node:fs:464:35)
    at /var/task/node_modules/pdf2json/pdfparser.cjs:1966:25
    at Array.reduce (<anonymous>)
    at Object.<anonymous> (/var/task/node_modules/pdf2json/pdfparser.cjs:1964:31)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at exports.b (/var/task/___vc/__launcher/chunk-BWVOFI3E.js:1:1211) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/var/task/node_modules/pdf2json/base/shared/util.js',
```

**Comment by FHoxhaDev** [comment]
Hi @modesty,

Is this an issue with Vercel or with the package itself? Is there any workaround to fix this error?

**Comment by kylerob** [comment]
It's an issue with Vercel and executing this code on the server side due to edge functions. The issue is all of [these files](https://github.com/modesty/pdf2json/blob/master/lib/pdf.js#L23), the first of which you're seeing an error for, [can't be read](https://github.com/modesty/pdf2json/blob/master/lib/pdf.js#L69) on the edge function because it hasn't been copied over by Vercel's edge function setup code. Vercel usually does a good job including what you need, but that's if you're importing code via module imports. This code is building up a string of all those file contents and eval'ing it. 

I have a [fork with a semi-working version](https://github.com/modesty/pdf2json/compare/master...kylerob:pdf2json:master) that creates the big import file in a separate step, and then imports it into the main files so Vercel includes everything. I'm working on fixing an issue with a canvas function that's defined in one file and used in another that's not working, but you can try it out if you want. If I get it working today I'll create a PR in the main repo to get it merged into the baseline.

**Comment by JinhooBong** [comment]
@kylerob by pointing my "pdf2json" in `package.json` will it point to your branch?

**Comment by FHoxhaDev** [comment]
Hi @kylerob ,

Thank you for your fix. Unfortunately, for me it didn't work. It is failing at my API call where it uploads the file, this is my endpoint:

https://gist.github.com/FHoxhaDev/59e2dc2b1cc134b6d064b96573a51145

and this is the error:

<img width="445" alt="image" src="https://github.com/modesty/pdf2json/assets/76188596/039e3678-80f2-4a2a-8ab3-e772ce84b4fe">

Locally it works but not on Vercel host.

**Comment by kylerob** [comment]
@FHoxhaDev Try forcing a redeploy on Vercel -- I ran into the same issue at first. Vercel seems to cache npm packages, because I've had trouble getting it to bring in my latest npm changes sometimes.

**Comment by adpadillar** [comment]
Hey @kylerob, thanks for your fix!

I just wanted to ask some quick questions. 
1. Any plans on making a PR with this fix at any point?
2. If not, any plan to publish a separate npm package with your fork

Cheers!

**Comment by modesty** [comment]
thanks to @kylerob. A more thorough fix plus appropriate tests are in [PR #338](https://github.com/modesty/pdf2json/pull/338), please review and test it out

**Comment by kylerob** [comment]
@modesty Thanks for bringing it home!

**Comment by modesty** [comment]
fixed in 3.1.2
